### PR TITLE
Case by owner remove usages

### PR DESCRIPTION
**Pull request by dannyroberts**
Continuing in the vein of https://github.com/dimagi/commcare-hq/pull/7014 and https://github.com/dimagi/commcare-hq/pull/7020

Notes:
- removes dca reports and thus the dca design doc—I believe there's no reindexing load that comes with removing a design doc (why would there be), but just wanted to flag that in case anyone knows otherwise
- This PR removes all calls to `case/by_owner` and `case/by_owner_lite`, using `hqcase/by_owner` (same but sorts by `domain` first) and where necessary `case/get_lite` instead.
- Those two views can be removed at any point now, but leaving that for a [separate PR](https://github.com/dimagi/commcare-hq/compare/remove-case-by_owner?expand=1) (to be opened once this is merged).

### CODE REVIEW

**Comment by dannyroberts** [comment]
@benrudolph or @czue fixed builds, ready for review. @kaapstorm buddy

File: corehq/apps/hqcase/management/commands/ptop_fast_reindex_cases.py [hunk 1]
```diff
@@ -7,5 +7,5 @@ class Command(ElasticReindexer):
     help = "Fast reindex of case elastic index by using the case view and reindexing cases"
 
     doc_class = CommCareCase
-    view_name = 'case/by_owner'
+    view_name = 'hqcase/by_owner'
```
Comment (benrudolph) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindex_cases.py]: interesting, was this just wrong?

Comment (dannyroberts) [inline corehq/apps/hqcase/management/commands/ptop_fast_reindex_cases.py]: No, it worked before too. The two views just have the same salient property of emitting each case exactly once, so they can be used in the same way here.

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py [hunk 2]
```diff
@@ -144,22 +146,22 @@ def __init__(self, restore_state, chunk_size=None):
...
     def batches(self):
-        for key in self.owner_keys:
+        for owner_id in self.owner_ids:
             batch = CaseSyncCouchBatch(
                 self.global_state,
                 self.domain,
                 self.last_synclog,
                 self.chunk_size,
-                key,
+                owner_id,
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py]: isn't this a different value from `key`? from what i understand, `key = [[owner_id, False]]` is that equivalent to just passing the `owner_id`?

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py [hunk 3]
```diff
@@ -278,9 +280,9 @@ class CaseSyncCouchBatch(CaseSyncBatch):
     Batch of case updates for cases 'owned' by the user.
     """
     def __init__(self, global_state, domain, last_sync, chunksize,
-                 startkey, case_sharing=False, startkey_docid=None):
+                 owner_id, case_sharing=False, startkey_docid=None):
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/batched.py]: ah ignore above comment, i see you've accounted for it

File: custom/_legacy/dca/_design/views/dca_collection_forms_by_case/map.js [hunk 4]
```diff
@@ -1,10 +0,0 @@
-function(doc) {
-    if (doc.doc_type === 'XFormInstance'
-        && doc.domain === 'dca-malawi'
-        && doc.xmlns === 'http://openrosa.org/formdesigner/53CCC67E-775F-4726-B187-BF558D40B679'){
-        var u = doc.form.meta;
-        var d = new Date(doc.form.meta.timeStart);
-
-        emit([doc.form.case.case_id, d.getUTCFullYear(), d.getUTCMonth(), d], null);
-    }
-}
```
Comment (benrudolph) [inline custom/_legacy/dca/_design/views/dca_collection_forms_by_case/map.js]: yay to deleting unused views!

**Comment by benrudolph** [comment]
cool, i'm good with this. do you want cory to look or should i go ahead and merge?

**Comment by dannyroberts** [comment]
The tests should be pretty good, and the changes I've made are of the "logically equivalent" sort, and when tests didn't pass originally I traced it back to problems with the test which is always validating. Feel free to merge!

**Comment by czue** [comment]
:+1: to not blocking on me :)

**Comment by snopoke** [comment]
Not sure if its necessary for this but I've been recording the git commit hashes of commits that remove custom code here: https://github.com/dimagi/commcare-hq/blob/master/custom/expired_domains.md

**Comment by dannyroberts** [comment]
@snopoke thanks for the reminder. I just committed a change to that file straight to master: https://github.com/dimagi/commcare-hq/commit/e8589e1cbd815670057915d3599c513af7295304

**Comment by snopoke** [comment]
:+1:

**Pull request merged**